src/Cli/dotnet/ProjectInstanceExtensions.cs

```diff
@@ -22,7 +22,12 @@ public static string GetProjectId(this ProjectInstance projectInstance)
 
         public static string GetDefaultProjectTypeGuid(this ProjectInstance projectInstance)
         {
-            return projectInstance.GetPropertyValue("DefaultProjectTypeGuid");
+            string projectTypeGuid = projectInstance.GetPropertyValue("DefaultProjectTypeGuid");
+            if (string.IsNullOrEmpty(projectTypeGuid) && projectInstance.FullPath.EndsWith(".shproj"))
```

Comment:
Ah, I see what you mean. Yeah this is a bit 'extra' but the user experience is nicer with this quirk in place 👍